### DISCUSSION TITLE
How can I find the version of a runner from a job?

### DISCUSSION THREAD

**Discussion by yeikel** [actions/runner#2838] [Q&A] [answered]
Given a job, how can I find the version of the current runner? 

I understand that the version is printed as part of the "set up job" step such as : 

> Current runner version: '2.303.0'

but I need it programmatically as part of one of my steps. 

Is there any environment variable or file storing it?

 I went over the code but it is [printing using a compiled constant](https://github.com/actions/runner/blob/3f5b81349912fa4fd3480171aa90eba300587131/src/Runner.Listener/Runner.cs#L369) that I can't access from a job

Note that I am not referring to https://github.com/actions/runner/discussions/1399 as that implies running the command from a runner

**Comment by artembilan** [comment]
I cannot answer to this, but I don't see a way to add comment.
I have exactly the same problem as I need such an info for some build report.

Thanks

**Comment by crohr** [comment]
@yeikel you should be able to get it from the archive name in `/opt/runner-cache/`

**Reply by artembilan** [reply]
@crohr ,

Would you mind to share that bash command we can perform to extract such a version?
It would be hard for us to try-n-fail until we find what exactly archive in that dir you are talking about.

Thanks

**Comment by crohr** [comment, marked as answer]
@artembilan here you are:

```bash
tarball=$(ls /opt/runner-cache/)
if [[ $tarball =~ ([0-9]+\.[0-9]+\.[0-9]+) ]]; then
  echo "Version: ${BASH_REMATCH[1]}"
else
  echo "Version not found."
fi
```

<img width="498" alt="image" src="https://github.com/actions/runner/assets/6114/651ed81c-358a-49b2-85da-d60ea0084050">

**Reply by tbutler-qontigo** [reply]
@crohr Will this also work on windows runners and self hosted?
It would be great to have this exposed as an environment variable

My use case is that we are using a mix of github standard runners and self-hosted runners and we want to periodically check that our runners are "in sync" with the standard runners.
We are checking certain tool versions like bash, pwsh, pester etc but we would also like to ensure we are using the same runner version too, so that we can minimise differences between standard and self-hosted runners and reduce the likelihood of getting different results depending on what type of runner was used.

Thanks
